### PR TITLE
Set severity of rule CA1870 to warning

### PR DESCRIPTION
**Pull request by MihaZupan**
Followup after https://github.com/dotnet/roslyn-analyzers/pull/6898, #91727

### CODE REVIEW

File: eng/CodeAnalysis.src.globalconfig [hunk 1]
```diff
@@ -489,6 +489,9 @@ dotnet_diagnostic.CA1868.severity = warning
 # CA1869: Cache and reuse 'JsonSerializerOptions' instances
 dotnet_diagnostic.CA1869.severity = warning
 
+# CA1870: Use a cached 'SearchValues' instance
```
Comment (stephentoub) [inline eng/CodeAnalysis.src.globalconfig:492]: Can you add it to the test file as well (as none or suggestion) for consistency?

**Review by stephentoub — APPROVED** [review]

**Comment by MihaZupan** [comment]
@MichalStrehovsky I'm seeing odd Segmentation fault failures in [`nativeaot.CustomMain`](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-92135-merge-ee7be0a1ae094dcfbb/nativeaot.CustomMain/1/console.ec9d4c1f.log?helixlogtype=result) and [`nativeaot.GenerateUnmanagedEntryPoints`](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-92135-merge-ee7be0a1ae094dcfbb/nativeaot.GenerateUnmanagedEntryPoints/1/console.479ad7ff.log?helixlogtype=result) -- does that look familiar / is there any chance they're related to this PR (it does touch nativeaot's corelib)?

**Comment by MichalStrehovsky** [comment]
> @MichalStrehovsky I'm seeing odd Segmentation fault failures in [`nativeaot.CustomMain`](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-92135-merge-ee7be0a1ae094dcfbb/nativeaot.CustomMain/1/console.ec9d4c1f.log?helixlogtype=result) and [`nativeaot.GenerateUnmanagedEntryPoints`](https://helixre107v0xdcypoyl9e7f.blob.core.windows.net/dotnet-runtime-refs-pull-92135-merge-ee7be0a1ae094dcfbb/nativeaot.GenerateUnmanagedEntryPoints/1/console.479ad7ff.log?helixlogtype=result) -- does that look familiar / is there any chance they're related to this PR (it does touch nativeaot's corelib)?

If you click through the test results, you'll find a link to how to download the crash dump for this (it's `runfo get-helix-payload -j 02391e98-2ed5-4279-97a4-f74c743cbe2c -w nativeaot.SmokeTests -o c:\hell`).

We're crashing on a stack overflow, the repeating part is:

```
 	Exceptions.exe!S_P_CoreLib_System_ThrowHelper__ThrowArgumentNullException() Line 298	Unknown
 	Exceptions.exe!S_P_CoreLib_System_MemoryExtensions__IndexOfAny_7<Char>() Line 2102	Unknown
 	Exceptions.exe!S_P_CoreLib_System_MemoryExtensions__ContainsAny_7<Char>() Line 496	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_General_Helpers__EscapeTypeNameIdentifier() Line 115	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__get_Name() Line 197	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_RuntimeNamedTypeInfo__get_FullName() Line 96	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__ToString() Line 147	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Exception__AppendExceptionStackFrame() Line 141	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Runtime_EH__AppendExceptionStackFrameViaClasslib() Line 270	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Runtime_EH__UpdateStackTrace() Line 881	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Runtime_EH__DispatchEx() Line 707	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Runtime_EH__RhThrowEx() Line 641	Unknown
 	Exceptions.exe!RhpThrowEx() Line 190	Unknown
 	Exceptions.exe!S_P_CoreLib_System_ThrowHelper__ThrowArgumentNullException() Line 298	Unknown

```

This by itself doesn't make much sense. The way we got into the repeating part is:

```
 	Exceptions.exe!S_P_CoreLib_System_ThrowHelper__ThrowArgumentNullException() Line 298	Unknown
 	Exceptions.exe!S_P_CoreLib_System_MemoryExtensions__IndexOfAny_7<Char>() Line 2102	Unknown
 	Exceptions.exe!S_P_CoreLib_System_MemoryExtensions__ContainsAny_7<Char>() Line 496	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_General_Helpers__EscapeTypeNameIdentifier() Line 115	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__get_Name() Line 197	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_RuntimeNamedTypeInfo__get_FullName() Line 96	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__ToString() Line 147	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_RuntimeTypeInfo__EstablishDebugName() Line 739	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__GetRuntimeNamedTypeInfo() Line 80	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_General_TypeUnifier__GetNamedType_0() Line 57	Unknown
 	Exceptions.exe!S_P_CoreLib_Internal_Reflection_Core_Execution_ExecutionDomain__GetNamedTypeForHandle() Line 106	Unknown
 	Exceptions.exe!S_P_Reflection_Execution_Internal_Reflection_Execution_ReflectionExecutionDomainCallbacksImplementation__GetNamedTypeForHandle() Line 40	Unknown
 	Exceptions.exe!S_P_CoreLib_Internal_Reflection_Core_NonPortable_RuntimeTypeUnifier__GetRuntimeTypeBypassCache() Line 91	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Type__GetTypeFromMethodTableSlow() Line 56	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Type__GetTypeFromMethodTable() Line 41	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Type__GetTypeFromHandle() Line 24	Unknown
 	Exceptions.exe!UInt32__CreateChecked<Char>() Line 630	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Buffers_SearchValues__TryGetSingleRange<Char>() Line 194	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Buffers_SearchValues__Create_0() Line 86	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_General_Helpers___cctor() Line 27	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Runtime_CompilerServices_ClassConstructorRunner__EnsureClassConstructorRun() Line 88	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Runtime_CompilerServices_ClassConstructorRunner__CheckStaticClassConstructionReturnGCStaticBase() Line 35	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_General_Helpers__EscapeTypeNameIdentifier() Line 115	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__get_Name() Line 197	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_RuntimeNamedTypeInfo__get_FullName() Line 96	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__ToString() Line 147	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_RuntimeTypeInfo__EstablishDebugName() Line 739	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_TypeInfos_NativeFormat_NativeFormatRuntimeNamedTypeInfo__GetRuntimeNamedTypeInfo() Line 80	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Reflection_Runtime_General_TypeUnifier__GetNamedType_0() Line 57	Unknown
 	Exceptions.exe!S_P_CoreLib_Internal_Reflection_Core_Execution_ExecutionDomain__GetNamedTypeForHandle() Line 106	Unknown
 	Exceptions.exe!S_P_Reflection_Execution_Internal_Reflection_Execution_ReflectionExecutionDomainCallbacksImplementation__GetNamedTypeForHandle() Line 40	Unknown
 	Exceptions.exe!S_P_CoreLib_Internal_Reflection_Core_NonPortable_RuntimeTypeUnifier__GetRuntimeTypeBypassCache() Line 91	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Type__GetTypeFromMethodTableSlow() Line 56	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Type__GetTypeFromMethodTable() Line 41	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Type__GetTypeFromHandle() Line 24	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Collections_Generic_Dictionary_2<System___Canon__System___Canon>___ctor_2() Line 65	Unknown
 	Exceptions.exe!S_P_CoreLib_System_Collections_Generic_Dictionary_2<System___Canon__System___Canon>___ctor_0() Line 42	Unknown
 	Exceptions.exe!S_P_CoreLib_System_AppContext__InitializeDataStore() Line 17	Unknown
 	Exceptions.exe!S_P_CoreLib_System_AppContext___cctor() Line 18	Unknown
```

The problem is that this PR is introducing use of reflection into the implementation of reflection. That doesn't lead to anything good.

**Comment by MihaZupan** [comment]
Thanks!
I've disabled this rule for that file in this case given it's the only problematic site.

**Pull request merged**